File: 06-2.py
#!/usr/bin/env python

lines = [row.strip() for row in open('06.input').readlines()]

time = int(lines[0].split(':')[1].replace(' ', ''))
record = int(lines[1].split(':')[1].replace(' ', ''))

wins = 0
for hold in range(time + 1):
    speed = hold
    distance = (time - hold) * speed
    if distance > record:
        wins += 1

print(wins)

File: 06-2.rb
#!/usr/bin/env ruby

lines = File.read('06.input').lines.map(&:strip)

time = lines[0].split(':')[1].delete(" ").to_i
record = lines[1].split(':')[1].delete(" ").to_i

wins = 0
0.upto(time) do |hold|
  speed = hold
  distance = (time - hold) * speed
  wins += 1 if distance > record
end

print wins, "\n"
